### ISSUE TITLE
Update Dart team-maintained code to use new part-of URI syntax

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#29592]
We have new part of syntax that takes a URI instead of an identifier because it's better. We should take advantage of that by going through all of the code the Dart team maintains — core libraries, tools, tests, packages, etc. — and switching to the new syntax where appropriate. Reasons:

* The new syntax is better. (If it isn't, why did we add it?)
* It will help dogfood it and flush out bugs in the implementation.
* It will give us first-hand experience with what it's like to use the new syntax.
* Users learn from our code, so it will teach them the way to write modern Dart.

This is the main tracking bug. Feel free to split it into sub bugs and farm them out to people as appropriate.

**Comment by lrhn** [comment]
It's unlikely that we will fix all of it. 
I'm going for the non-web platform libraries, but I won't try touching dart:html because it's generated files. I think the dart4web people should handle that. Also, internal VM libraries like the observatory should probably be handled by the VM team.

Luckily, most packages are part-free, so we do have that going for us.

**Comment by munificent** [comment]
>  I think the dart4web people should handle that. Also, internal VM libraries like the observatory should probably be handled by the VM team.

Would you mind filing sub-issues for these and assigning them to the relevant people. An email to the team saying, "Hey, there's this new syntax. Can you migrate your code to it?" wouldn't hurt either.

**Comment by jacehensley-wf** [comment]
I noticed that the changelog entry for the new URI syntax was removed. Does that mean it didn't make 1.23.0?

https://github.com/dart-lang/sdk/commit/9add06fe7f835ec2a95175184cdcaa3d72f7c18d

**Comment by munificent** [comment]
That's right. Technically, some of the tools in 1.23 will handle the syntax, but they don't all support it. (In particular, I know dartfmt doesn't.) It will be officially supported in 1.24.

**Comment by lrhn** [comment]
Dart core platform libraries blocked by #29598.

**Comment by dgrove** [comment]
Is this actually required for 1.24? If not, please move to 1.25 .

**Comment by dgrove** [comment]
I believe this should have been moved to 1.25. #29598 has already been moved to 1.25.

**Comment by lrhn** [comment]
This change hit some snags with Fasta, and it's not a priority to fix it right now. I'll keep this issue open in case we want to revisit it later.

**Comment by kevmoo** [comment]
@munificent – maybe make this a lint? We could add it to `pkg:pedantic` to prune the remaining offenders.

**Comment by munificent** [comment]
Linting SGTM.
